vlm/src/main/scala/WarpGeoTiffRasterSource.scala

```diff
@@ -0,0 +1,58 @@
+package geotrellis.contrib.vlm
+
+import geotrellis.vector._
+import geotrellis.raster._
+import geotrellis.raster.reproject._
+import geotrellis.raster.resample._
+import geotrellis.proj4._
+import geotrellis.raster.io.geotiff.MultibandGeoTiff
+import geotrellis.raster.io.geotiff.reader.GeoTiffReader
+import cats.effect.IO
+
+import java.net.URI
+import java.nio.file.Paths
+
+import geotrellis.util.{FileRangeReader, RangeReader, StreamingByteReader}
+
+case class WarpGeoTiffRasterSource(
```

Comment:
I think it'd be nice to have just a single, generic `WarpRasterSource`; but I also think we'd have to decide on how we're going to implement the other features before deciding on the `WarpRasterSource` API.

Reply:
👍